packages/clay-navigation-bar/src/index.tsx

```diff
@@ -0,0 +1,167 @@
+/**
+ * © 2019 Liferay, Inc. <https://liferay.com>
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import classNames from 'classnames';
+import React from 'react';
+
+import ClayIcon from '@clayui/icon';
+import ClayLink from '@clayui/link';
+import warning from 'warning';
+
+import {useTransition} from './Hooks';
+
+interface ItemProps extends React.HTMLAttributes<HTMLLIElement> {
+	/**
+	 * Determines the active state of an dropdown list item.
+	 */
+	active?: boolean;
+
+	/**
+	 * Children elements received from ClayNavigationBar.Item component.
+	 */
+	children: React.ReactElement;
+}
+
+interface Props extends React.HTMLAttributes<HTMLDivElement> {
+	/**
+	 * Children elements received from ClayNavigationBar component.
+	 */
+	children: React.ReactElement<ItemProps>[];
+
+	/**
+	 * Determines the style of the Navigation Bar
+	 */
+	inverted?: boolean;
+
+	/**
+	 * Path to the location of the spritemap resource.
+	 */
+	spritemap: string;
+
+	/**
+	 * Set up dropdown's trigger label.
+	 */
+	triggerLabel: string;
+}
+
+type ItemType = React.FunctionComponent<ItemProps>;
+
+const Item: ItemType = ({active, children, className, ...otherProps}) => {
+	return (
+		<li
+			{...otherProps}
+			className={classNames('nav-item', {
+				active,
+				className,
+			})}
+		>
+			{React.Children.map(
+				children,
+				(child: React.ReactElement<ItemProps>, index) => {
+					return React.cloneElement(child, {
+						...child.props,
+						className: classNames(
+							'nav-link',
+							child.props.className,
+							{
+								active,
+							}
+						),
+						key: index,
+					});
+				}
+			)}
+		</li>
+	);
+};
+
+const ClayNavigationBar: React.FunctionComponent<Props> & {
+	Item: ItemType;
+} = ({
+	children,
+	className,
+	inverted = false,
+	spritemap,
+	triggerLabel,
+	...otherProps
+}) => {
+	const [visible, setVisible] = React.useState(false);
+	const contentRef = React.useRef<HTMLDivElement>(null);
+	const [
+		transitioning,
+		handleTransitionEnd,
+		handleClickToggler,
+	] = useTransition(visible, setVisible, contentRef);
+
+	const activeElementsCount = children.filter(child => child.props.active)
+		.length;
+
+	warning(
+		!(activeElementsCount > 1),
+		`You passed ${activeElementsCount} active props to ClayNavigationBar children.`
+	);
+
+	warning(
```

Comment:
Is this necessary since we require it via typescript params?

Reply:
Nope. Removing!